mycity/mycity/utilities/arcgis_utils.py

```diff
@@ -0,0 +1,283 @@
+    :return Dictionary of updated request body parameters
+    """
+    updated_params = {}
+    for key in params:
+        value = params[key]
+        updated_params[key] = (None, str(value))
+    logger.debug("Updated Parameters: {}".format(str(updated_params)))
+    return updated_params
+
+
+
+
+def _format_float(input_float):
+    """
+    Takes a float and returns a formatted String
+
+    :param input_float: Floating point number
+    :return: String from a floating point number,
+        rounded to two decimals
+    """
+
+    rounded = round(input_float, 2)
+    as_string = str(rounded)
+    return as_string
+
+def _post_request(url, params, headers):
+    """
+    Utilizes requests module to formulate
+    HTTP POST request over the network
+
+    :param url: String representing base URL of request
+    :param params: String or Dictionary containing parameters
+    :param headers: Dictionary containing headers
+    :return: request.Response object
+    """
+    logger.debug("URL: {}, Params: {}, Headers: {}".format(url, str(params), str(headers)))
+
+    session = requests.Session()
+    request = requests.Request("POST", url, data=params, headers=headers)
+    prepared_request = request.prepare()
+    response = session.send(prepared_request)
+    return response
+
+
+
+def geocode_address_candidates(input_address):
+    """
+    Finds candidate addresses from ArcGIS Geocoding service
+    based on input address string
+
+    :param input_address: String of address to be geocoded
+    :return: JSON object containing geocoded address candidates
+    """
+    logger.debug("Input Address: {}".format(input_address))
+
+    base_url = "https://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/findAddressCandidates"
+    params = {
+            "f": "json",
+            "singleLine": input_address,
+            "outFields":"Match_addr,Addr_type"
+            }
+    headers = {
+            'cache-control': "no-cache"
+            }
+
+    response = requests.request("GET", base_url, params=params, headers=headers)
+    if response.status_code != 200:
+        return None
+    else:
+        return response.json()
+
+
+def select_top_address_candidate(geocode_candidate_response_json):
+    """
+    Selects the geocoded address candidate with the highest score
+
+    :param geocode_candidate_response_json: JSON object containing address candidates
+        returned from the ArcGIS Geodcoding API
+    :return: Dict containing address string and
+        location information ((X, Y) coordinates)
+    """
+
+    candidates = geocode_candidate_response_json['candidates']
+    logger.debug("Geocode Candidates: " + str(candidates))
+
+    if len(candidates) == 0:
+        return None
+    else:
+        top_score = -1.0
+        top_candidate = None
+        for candidate in candidates:
+            if candidate['score'] > top_score:
+                top_candidate = candidate
+                top_score = candidate['score']
```

Comment:
You might be able to use a custom `max()` function here. Something like
`
top_candidate = max(candidates, key=lambda candidate : candidate['score'])
`